src/services/redisSessionStorage.ts

```diff
@@ -256,42 +251,38 @@ export default class RedisSessionStorage extends RedisStorageUtils
   /**
    * {@inheritDoc}
    */
-  public async del(
-    sessionToken: SessionToken,
-    walletToken: WalletToken,
-    myPortalToken?: MyPortalToken,
-    bpdToken?: BPDToken
-  ): Promise<Either<Error, boolean>> {
-    const user = await this.loadSessionBySessionToken(sessionToken);
-    if (isLeft(user)) {
-      return left(user.value);
-    }
+  public async del(user: User): Promise<Either<Error, boolean>> {
     const tokens: ReadonlyArray<string> = [
-      `${sessionKeyPrefix}${sessionToken}`,
-      `${walletKeyPrefix}${walletToken}`,
-      ...(myPortalToken ? [`${myPortalTokenPrefix}${myPortalToken}`] : []),
-      ...(bpdToken ? [`${bpdTokenPrefix}${bpdToken}`] : [])
+      `${sessionKeyPrefix}${user.session_token}`,
+      `${sessionInfoKeyPrefix}${user.session_token}`,
+      `${walletKeyPrefix}${user.wallet_token}`,
+      ...(UserV2.is(user)
+        ? [`${myPortalTokenPrefix}${user.myportal_token}`]
+        : []),
+      ...(UserV3.is(user) ? [`${bpdTokenPrefix}${user.bpd_token}`] : [])
```

Comment:
a more reusable way would be to have a `getUserTokens(user: User)` method 
that returns a StrMap (<token_name> -> {prefix: <prefix>, value: <value>, ttl: ... }) eventually using an [exhaustive switch](https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript) on UserV2.is(...) 

usage here would be: 
```
collect(getUserTokens(user), ((_, {prefix, value}) => `${prefix}${value}`)
```